consensus/blake3pow/poem.go

```diff
@@ -203,3 +264,17 @@ func (blake3pow *Blake3pow) CalcRank(chain consensus.GenesisReader, header *type
 
 	return 0, nil
 }
+
+func (blake3pow *Blake3pow) CheckIfValidWorkShare(workShare *types.WorkObjectHeader) bool {
+	// Extract some data from the header
+	diff := new(big.Int).Set(workShare.Difficulty())
+	c, _ := mathutil.BinaryLog(diff, mantBits)
+	workShareThreshold := c - params.WorkSharesThresholdDiff
```

Comment:
Maybe check that c >= params.WorkSharesThersholdDiff?

Reply:
i guess we can check, but the diff needs to be less than 8 for this to crash, i think min diff we allow is wayy greater than that, if you want, i can add that check

Reply:
I think any time we subtract any numbers we should check for negative/underflow

Reply:
adding a check